doc/using/overrides.chapter.md

```diff
@@ -40,6 +40,8 @@ import pkgs.path { overlays = [ (self: super: {
 
 In the first example, `pkgs.foo` is the result of a function call with some default arguments, usually a derivation. Using `pkgs.foo.override` will call the same function with the given new arguments.
 
+Many packages, like the `foo` example above, provide package options with default values in their arguments, to facilitate overriding. Because it's not usually feasible to test that packages build with all combinations of options, you might find that a package doesn't build if you override options to non-default values. Package maintainers are not expected to fix arbitrary combinations of options, so if you want to use non-default package options and find that it doesn't work, don't be surprised if you're asked to fix it yourself.
```

Comment:
```suggestion
Many packages, like the `foo` example above, provide package options with default values in their arguments, to facilitate overriding.
Because it's not usually feasible to test that packages build with all combinations of options, you might find that a package doesn't build if you override options to non-default values.
Package maintainers are not expected to fix arbitrary combinations of options, so if you want to use non-default package options and find that it doesn't work, please submit a fix with a regression test.
```

I suppose we can phrase this constructively and make it actionable. Also it rubs me the wrong way to warn people in the manual about maintainers telling them to go fix their own problems.

Reply:
> warn people in the manual about maintainers telling them to go fix their own problems.

Where should we document what people should expect and not expect from package maintainers, then?

Reply:
BTW: I wrote this all on one line because that's what the rest of the file does.

Reply:
I'm not sure your suggestion conveys what I think it's important to convey — this came out of a conversation I had on Matrix with @k900 about avoiding having package options at all, in case it prompts too many issues about obscure combinations of options not working.  So just telling people to submit fixes doesn't really help avoid that.

Reply:
Okay so do you rather want to say "this stuff exists but doesn't really scale, so don't expect all the options to work and don't push to add more"?

And where to add what to expect from maintainers... generally I'd say "in the contributor guide", but still I doubt we want to convey "if you ask this you'll get snarky responses" and would instead suggest phrasing it as a recommendation for maintains on how to deal constructively with certain situations, or even better: describe the desired end state of the system and thus nudge people to behave such that we eventually arrive at that state.

Reply:
> Okay so do you rather want to say "this stuff exists but doesn't really scale, so don't expect all the options to work and don't push to add more"?

Yes to not expecting them to work, but adding more is fine — the risk is that we end up with options that nobody cares about, and so nobody is putting any effort into keeping them working.

> describe the desired end state of the system and thus nudge people to behave such that we eventually arrive at that state.

For me, the desired end state of the system is that package maintainers are open to adding and keeping plenty of options, because they don't fear being held responsible for them working, or having to repeatedly deal with issues about them not working, which is emotionally taxing when you keep having to let people down because they didn't know what to expect.  Not doing anything to manage expectations reach making Nixpkgs less useful, because maintainers will proactively remove options to avoid being inundated with this sort of issue.

Reply:
Okay, how about this then:

```suggestion
Many packages, like the `foo` example above, provide package options with default values in their arguments, to facilitate overriding.
Because it's not usually feasible to test that packages build with all combinations of options, you might find that a package doesn't build if you override options to non-default values.

Package maintainers are not expected to fix arbitrary combinations of options.
If you find that something doesn't work, please submit a fix, ideally with a regression test.
If you want to ensure that things keep working, consider [becoming a maintainer](https://github.com/NixOS/nixpkgs/tree/master/maintainers) for the package.
```

Reply:
Ooh, thanks for the an alternative proposal. :)  I was worried this wasn't going to go anywhere.

In https://github.com/NixOS/nixpkgs/pull/331999 I agreed to maintain specifically `pipewire.override { enableSystemd; }`, but not pipewire overall.  So we could mention that possibility, but on the other hand I think this might be the only occurrence of it, so maybe it's better not to suggest it unless it becomes a more widespread practice.

Does your suggested change address your concern that you said rubbed you the wrong way before?  I think maybe I misunderstood it at the time — at the time I thought you didn't want the manual to warn about imperfect package maintenance, but now re-reading it I think maybe you meant not wanting the manual to say that people should expect an unpleasant reaction from maintainers?

(To be clear, I never wanted to express that people should express a hostile, "snarky", etc. reaction from maintainers, just that they should be aware it's reasonable for the maintainer not to want to fix all the option combinations.)

Reply:
Yeah, I'd be happy with the  suggestion as given. No problem with being honest about imperfect maintenance!